core/src/main/scala/sttp/tapir/server/interpreter/BodyListener.scala

```diff
@@ -0,0 +1,11 @@
+package sttp.tapir.server.interpreter
+
+trait BodyListener[F[_], B] {
+  def onComplete(body: B)(cb: => F[Unit]): F[B]
```

Comment:
maybe it would be useful to also have an `onFailure` callback - so that we can also capture situations, where producing the body fails (which results in a broken connection, as the headers are already sent out)

this could be a separate callback, or a single one taking a `Try[Unit]` parameter. Maybe see the `Future`/`Try` API for an inspiration?